packages/core/utils/lib/sanitize/visitors/remove-password.js

```diff
@@ -1,7 +1,7 @@
 'use strict';
 
-module.exports = ({ key, attribute }, { remove }) => {
+module.exports = ({ data, key, attribute }) => {
   if (attribute?.type === 'password') {
-    remove(key);
+    delete data[key];
```

Comment:
for some reason doing it like this breaks some tests 🤔  (sanitize-query.test.api.js )

Reply:
I don't think we should delete the key directly, as we are using this visitor in more places (traverseQueryFilters) https://github.com/strapi/strapi/blob/71fab844d3be6eed39003b77bc7bbd649f4b5b94/packages/core/utils/lib/traverse/query-fields.js#L22-L24

In there, we are not deleting the key from an object but something else.

I just tested and this does not affect performance, as it's only executed for password fields ( and private) which are minimal compared to the number of attributes in a normal CT.

wdyt if we rever this @Convly ?

Reply:
Yep we should keep the utils logic abstracted and implemented by the caller